de.metas.business/src/main/java/org/adempiere/mm/attributes/api/impl/AttributeSetInstanceBL.java

```diff
@@ -275,6 +278,43 @@ public I_M_AttributeInstance getCreateAttributeInstance(final I_M_AttributeSetIn
 		return instanceNew;
 	}
 
+	@Override
+	public void setAttributeInstanceValue(@NonNull final I_M_AttributeSetInstance asi, @NonNull final I_M_Attribute attribute, @NonNull final Object value)
+	{
+		final I_M_AttributeInstance attributeInstance = Services.get(IAttributeDAO.class).retrieveAttributeInstance(asi, attribute.getM_Attribute_ID());
+		if (attributeInstance == null)
+		{
+			throw new AdempiereException("instanceExisting not exists");
+		}
+
+		final String attributeValueType = attribute.getAttributeValueType();
+		if (X_M_Attribute.ATTRIBUTEVALUETYPE_Date.equals(attributeValueType))
+		{
+			attributeInstance.setValueDate(Env.parseTimestamp(value.toString()));
+		}
+		else if (X_M_Attribute.ATTRIBUTEVALUETYPE_List.equals(attributeValueType))
+		{
+			attributeInstance.setValue(value.toString());
+
+			final I_M_AttributeValue attributeValue = Services.get(IAttributeDAO.class).retrieveAttributeValueOrNull(attribute, value.toString());
+			attributeInstance.setM_AttributeValue(attributeValue);
```

Comment:
i think setting the M_AttributeValue_ID is not needed. see de.metas.modelvalidator.M_AttributeInstance.updateAttributeValueIfList(I_M_AttributeInstance)